### PR TITLE
Add unit tests for the Step component

### LINKED ISSUE #123: Add unit and e2e tests for the Step component
Unit and e2e tests need to be written for the Step component. Unit tests are done with [Jest](https://jestjs.io/), and e2e tests are done with [nightwatch](https://nightwatchjs.org/).

Please remember to test the following things:
- That individual parts of the component are present when appropriate. (unit and e2e)
- That any computed props and methods work properly, if there are any. (unit)
- Any common interactions between the user and component, if there are any. (e2e)
- Any other functionality unique to the component being tested!

### Additional Context
- [./src/components/Step.vue](https://github.com/creativecommons/cc-chooser/blob/master/src/components/Step.vue)
- [This repo's testing README](https://github.com/creativecommons/cc-chooser/blob/master/tests/README.md)
- [Vue's guide on unit testing](https://vuejs.org/v2/guide/unit-testing.html)
- [Vue's guide on unit testing with VueX](https://vue-test-utils.vuejs.org/guides/using-with-vuex.html)

### PR DESCRIPTION
**Pull request by akmadian**
## Fixes

Fixes #123 by @akmadian 

## Description

This PR adds unit tests for the Step component.

## Technical details

 - This PR does not test every possible permutation of props because simply testing the logic in this particular component should cover almost all cases assuming the locale file has the proper schema.
 - This PR does not include e2e tests because just about all permutations of step functionality are covered in the Stepper component's e2e tests.

## Tests

Run `npm run test`

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: tests/unit/specs/components/Step.spec.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        Vue.use(VueI18n)
+        const i18n = new VueI18n({
+            locale: 'en',
+            fallbackLocale: 'en',
+            messages: {}
+        })
+
+        config.mocks.i18n = i18n
+
+        config.mocks.$t = key => key
```
Comment (obulat) [inline tests/unit/specs/components/Step.spec.js:27]: If we return key, it is added to html as is (eg. `stepper.SA.not-selected`), which is enough for testing.

File: tests/unit/specs/components/__snapshots__/Step.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      <b-radio-stub value="no" nativevalue="yes"><span class="vocab-body body-normal">stepper.yesstepper.SA.selected</span></b-radio-stub>
+    </div>
+    <div class="field selected">
+      <b-radio-stub value="no" nativevalue="no"><span class="vocab-body body-normal">stepper.nostepper.SA.not-selected</span></b-radio-stub>
+    </div>
+  </div>
+</div>
+`;
+
+exports[`Step.vue Check if the Step.vue component has the expected UI if disabled due is set 1`] = `
```
Comment (obulat) [inline tests/unit/specs/components/__snapshots__/Step.spec.js.snap:16]: Oops, this doesn't work as it is supposed to. I will check that.

**Review by obulat — APPROVED** [review]
Great! 
I've added some snapshot testing for permutation of props, because the layout of the component changes accordingly. 
Also, I've finally found the best way of testing with i18n in chooser: by mocking it and returning the key itself, so that the key is used as the text for html. 
I hope it's ok with you, @akmadian .

File: tests/unit/specs/components/__snapshots__/Step.spec.js.snap (hunk 2 above)
Comment (obulat) [reply tests/unit/specs/components/__snapshots__/Step.spec.js.snap:16]: Fixed in the last commit.

**Comment by akmadian** [comment]
Yes that looks great! Thanks for catching and fixing those things : ) I'll merge

**Pull request merged**